### PR TITLE
feat: hook changes

### PR DESCRIPTION
**Pull request by kopy-kat**
- [x] initial changes
- [ ] pass correct values to postHook

### CODE REVIEW

**Review by livingrockrises — APPROVED** [review]

**Review by kristofgazso — APPROVED** [review]
LGTM

File: src/core/HookManager.sol [hunk 1]
```diff
@@ -22,14 +22,24 @@ abstract contract HookManager {
     error HookPostCheckFailed();
     error HookAlreadyInstalled(address currentHook);
 
-    modifier withHook() {
+    function _preCheck() internal returns (address hook, bytes memory hookData) {
         address hook = _getHook();
```
Comment (livingrockrises) [inline src/core/HookManager.sol:26]: since type is already declared in return 
change this 
address hook -> hook
same with hookData

**Review by livingrockrises — COMMENTED** [review]
LGTM

File: src/core/HookManager.sol (hunk 1 above)
Comment (kopy-kat) [reply src/core/HookManager.sol]: good shout, done

**Pull request merged**